### PR TITLE
KOGITO-7633: SWF Viewer - Usability improvements for mediators (Preview Window)

### PR DESCRIPTION
**Pull request by Josephblt**
**JIRA:** [KOGITO-7633 - SWF Viewer - Usability improvements for mediators (Preview Window)](https://issues.redhat.com/browse/KOGITO-7633)

This adds a Preview Window tool. This window facilitates navigation when zoomed in.
Useful for large diagrams.
Just click the preview button to open/close it.

**VS Code Extension:** [Serverless Workflow](https://drive.google.com/file/d/13VORKXu5zxrMWEBs_TpJTLFyZR9GbBO2/view?usp=share_link)

### CODE REVIEW

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/components/mediators/ZoomLevelSelectorPresenter.java [hunk 1]
```diff
@@ -272,6 +281,10 @@ private void updateSelectorLevel(final double level) {
         reposition();
     }
 
+    private void showPreview(final double x, final double y) {
+
```
Comment (domhanak) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/components/mediators/ZoomLevelSelectorPresenter.java]: missing method body?

**Review by domhanak — CHANGES_REQUESTED** [review]
There is one blocker imho, not sure thought, the JIRA itself has no acceptance criteria so I do not know what is the target of testing.
I check the preview button from widget and the problem is that when I click `Preview` & hover away from it, the widget disappears, so I am not able to reset the viewer and a grey border is shown.

**Review by romartin — CHANGES_REQUESTED** [review]
Hey @Josephblt 

Code looks good!

I am also quickly testing the UI and 2 issues, for now. Will continue testing. Can you please take a look at those?

1. Use come icon for the preview button (instead of text)
2. Once I toggle the preview, it appears floating near to the top left corner... but it should be floating on top of the mediators widget (buttons). It should look like a expand/collapse panel. See screenshot:

Screenshot from 2022-09-19 15-58-07

Thanks!

**Comment by romartin** [comment]
Hey @Josephblt 

Thanks for the fixes and improvements. I still found a **few things to do**, let me know what you think:

1. The preview window is too big IMO, maybe we can do it smaller. There is lot of unused free space there.... see screenshot below
2. Once decided the right preview size (point above), can we then fit the workflow into that size, in the preview? Similar as we do with the main canvas once opening a workflow
3. The preview window overlaps the buttons bar on bottom (zoom, fit, etc), and it should not, just because features from icons should be also available meanwhile preview is opened
4. The preview window gets closed just on mouse exit the area. I think the expected behavior, at least what other tools also do, is to open the preview once clicking on the icon, and it will REMAIN OPEN until the user clicks again on the icon. This way it allows to interact with the workflow meanwhile having the preview opened, so users can quickly navigate using the preview as well
5. Issue -> The preview area (draggable) is not properly located as the main canvas is, see the following screenshot, where the preview is on an "empty" area on east side, but the main canvas show that there exists shapes there. This can be due to the size of the preview, in relation to the workflow size, we have to take a look at the impl here and see why the red rectangle does not follows the main canvas scrolling

Screenshot from 2022-10-10 22-39-03

Thanks!

**Comment by romartin** [comment]
@Josephblt any updates on this?

**Comment by Josephblt** [comment]
Hello @romartin, @handreyrc. Here is the PR for the preview.

**Comment by romartin** [comment]
hey @Josephblt , is the PR ready for review again? Please let us know once ready, so we can start it. Thanks!

**Comment by romartin** [comment]
@romartin  @domhanak ready for review again!

**Review by romartin — APPROVED** [review]
Hey @Josephblt 

Thanks for the updates. All issues I reported previously are being fixed, great job, it works nice!

**Comment by romartin** [comment]
Please @domhanak 

Missing QE reivew here. Thanks!

**Comment by domhanak** [comment]
@Josephblt @romartin Found an issue.

Steps:
1. Open diagram and zoom in to enable the preview window. Open it.
2. Rename a state so that diagram gets broken.
3. Observe the preview window going grey, while main canvas still is not grey.
4. Repair the damage. Observe that the window is still grey.

Expected behavior: (imho) it should reset the preview when diagram gets fixed.

Workaround: Reopen the preview window, you may have to zoom in to enable the button.

 Imho we can report it as minor issue which will most likely not be fixed due to its "edge-caseness" and very low severity. LMK.

**Comment by romartin** [comment]
thanks @domhanak . @Josephblt can you please tak ea look?

**Comment by Josephblt** [comment]
Hello @domhanak, @romartin. I updated the PR and .vsix. Ready for another round of tests. Thanks.

**Comment by domhanak** [comment]
@Josephblt Thanks! Will take a look asap.

**Review by domhanak — APPROVED** [review]
Thanks! Works well.

**Review by romartin — APPROVED** [review]

**Pull request merged**